client/checkout/classic/upe.js

```diff
@@ -0,0 +1,458 @@
+				messageWrapper +
+				'</div>'
+		);
+		$container
+			.find( '.input-text, select, input:checkbox' )
+			.trigger( 'validate' )
+			.blur();
+
+		let scrollElement = $( '.woocommerce-NoticeGroup-checkout' );
+		if ( ! scrollElement.length ) {
+			scrollElement = $container;
+		}
+
+		$.scroll_to_notices( scrollElement );
+		$( document.body ).trigger( 'checkout_error' );
+	};
+
+	/**
+	 * Mounts Stripe UPE element if feature is enabled.
+	 *
+	 * @param {boolean} isSetupIntent {Boolean} isSetupIntent Set to true if we are on My Account adding a payment method.
+	 */
+	const mountUPEElement = function ( isSetupIntent = false ) {
+		// Do not mount UPE twice.
+		if ( upeElement || paymentIntentId ) {
+			return;
+		}
+		const intentAction = isSetupIntent
+			? api.initSetupIntent()
+			: api.createIntent();
+
+		intentAction
+			.then( ( response ) => {
+				// I repeat, do NOT mount UPE twice.
+				if ( upeElement || paymentIntentId ) {
+					return;
+				}
+
+				const { client_secret: clientSecret, id: id } = response;
+				paymentIntentId = id;
+
+				hiddenElementsForUPE.init();
+				const appearance = getAppearance();
+				hiddenElementsForUPE.cleanup();
+				const businessName = getConfig( 'accountDescriptor' );
+
+				upeElement = elements.create( 'payment', {
+					clientSecret,
+					appearance,
+					business: { name: businessName },
+				} );
+				upeElement.mount( '#wcpay-upe-element' );
+				upeElement.on( 'change', ( event ) => {
+					isUPEComplete = event.complete;
+				} );
+			} )
+			.catch( ( error ) => {
+				showError( error.message );
+				const gatewayErrorMessage =
+					'<div>An error was encountered when preparing the payment form. Please try again later.</div>';
+				$( '.payment_box.payment_method_woocommerce_payments' ).html(
+					gatewayErrorMessage
+				);
+			} );
+	};
+
+	// Only attempt to mount the card element once that section of the page has loaded. We can use the updated_checkout
+	// event for this. This part of the page can also reload based on changes to checkout details, so we call unmount
+	// first to ensure the card element is re-mounted correctly.
+	$( document.body ).on( 'updated_checkout', () => {
+		// If the card element selector doesn't exist, then do nothing (for example, when a 100% discount coupon is applied).
+		// We also don't re-mount if already mounted in DOM.
+		if (
+			$( '#wcpay-upe-element' ).length &&
+			! $( '#wcpay-upe-element' ).children().length &&
+			isUPEEnabled &&
+			! upeElement
+		) {
+			mountUPEElement();
+		}
+	} );
+
+	if (
+		$( 'form#add_payment_method' ).length ||
+		$( 'form#order_review' ).length
+	) {
+		if (
+			$( '#wcpay-upe-element' ).length &&
+			! $( '#wcpay-upe-element' ).children().length &&
+			isUPEEnabled &&
+			! upeElement
+		) {
+			const useSetUpIntent = $( 'form#add_payment_method' ).length;
+			mountUPEElement( useSetUpIntent );
+		}
+	}
+
+	/**
+	 * Submits checkout form via AJAX to create order and uses custom
```

Comment:
Minor note, I think this comment just needs to be updated to mention the add payment flow instead of the checkout flow.

Reply:
Thanks for noticing! I changed it and took the opportunity to improve the name of a var that was not very clear.